packages/frontend/src/state/store.ts

```diff
@@ -36,6 +39,20 @@ export const MyShifts = selector({
   },
 });
 
+export const MyRolesState = selector<RoleConfig[]>({
+  key: 'myRoles',
+  get: async ({ get }) => {
+    const thisUser = get(UserState);
+    if (thisUser) {
+      const shiftClient = new BackendRoleClient(config.BackendURL);
+      const roles = await shiftClient.GetMyRoles();
+
+      return roles;
+    }
+    return [];
```

Comment:
Personaly preference _ I would invert the conditional here to "early return"

Kinda what [this person](https://web.archive.org/web/20180326123017/https://blog.timoxley.com/post/47041269194/avoid-else-return-early) means when they say "Try keep the 'meat' of your method at the lowest indentation level." It let's the reader mentally prune one of the branches

Reply:
agreed. I refactored the other bits of logic in this file to follow that as well.